finality-aleph/src/network/manager/session.rs

```diff
@@ -379,8 +380,8 @@ mod tests {
     async fn fails_to_create_with_non_unique_peer_id() {
         let mut crypto_basics = crypto_basics(NUM_NODES).await;
         let addresses = vec![
-            MockMultiaddress::random_with_id(MockPeerId::random()),
-            MockMultiaddress::random_with_id(MockPeerId::random()),
+            random_identity().0[0].clone(),
```

Comment:
maybe create a function `random_address` that does exactly that?

Reply:
Done, also fixed some of the clippy things – the ones I could reasonably consider parts of this PR.